### DISCUSSION TITLE
How can I add a new socket to the NODE GROUP FUNCTION Input node from a Python script?

### DISCUSSION THREAD

**Discussion by Sdphu** [bnpr/Malt#503] [Q&A] [answered]
I'm working on a Python script, and I'm using the node-groups-latest version.
Is there a way to execute the function that is triggered by the "+" button in the Properties panel from the script?

![001](https://github.com/bnpr/Malt/assets/89537457/f9fcb558-8170-4423-b873-cae0fe9e251d)

**Comment by pragma37** [comment, marked as answer]
Not directly, since it's a nested function, maybe I should move it to the class scope.
You can look at the source for reference: https://github.com/bnpr/Malt/blob/dfdb64d371fadb2170f7ffc4bcc58bf301736bc9/BlenderMalt/MaltNodes/Nodes/MaltIONode.py#L188

It boils down to something like this:
```python
parameter = node.get_custom_parameters().add()
param.graph_type = node.id_data.graph_type
param.is_output = node.is_output
param.io_type = node.io_type
param.name = "My Custom Parameter"
param.parameter = "vec3"
```
